pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -94,6 +94,13 @@ type CephClusterList struct {
 }
 
 // ClusterSpec represents the specification of Ceph Cluster
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.mon)))))",message="external mode enabled cluster cannot have mon field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.dashboard)))))",message="external mode enabled cluster cannot have dashboard field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.monitoring)))))",message="external mode enabled cluster cannot have monitoring field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.network)))))",message="external mode enabled cluster cannot have network field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.storage)))))",message="external mode enabled cluster cannot have storage field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.disruptionManagement)))))",message="external mode enabled cluster cannot have disruptionManagement field in CR"
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.mgr)))))",message="external mode enabled cluster cannot have mgr field in CR"
```

Comment:
Is it possible to reorder the params so that message comes first? When reading these in the future, it might be easier to read the intent first, followed by the rule definition after.

IMO, that makes it slightly easier to see at a glance what validations are implemented to see if new ones need to be added. It also makes it slightly easier to find a specific validation to edit.

```
// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.storage)))))",message="external mode enabled cluster cannot have storage field in CR"
```

```
// +kubebuilder:validation:XValidation:message="external mode enabled cluster cannot have storage field in CR",rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.storage)))))"
```

Reply:
I'm thinking of removing this validating rule since this is causing the `MultiCluster`  CI to fail because there we test external cluster and define `mon` and other fields.  Also,  it may cause issues with consumer and provider mode type clusters. @Madhu-1 any idea if this will affect you?

Reply:
in provider model its a internal ceph cluster with Host Network, nothing extra

Reply:
The external cluster shouldn't be defining the mons or other fields, so the rules should be fine. Note that two cephcluster CRs are created in the multicluster test, one for external and one for internal. The external one should not be defining mons or other ceph settings.

Reply:
The conversation here doesn't answer the question I posed initially. Is it possible to reorder the items on each line so that the `message=` appears before `rule=`? 

I think this will make it easier to read at a glance what the rules' intentions are. Otherwise, we will only be reading the really arcane rule syntax and have to scroll an arbitrary distance to the right to read what it's supposed to do.

Reply:
Yes, we can have `message=` appear first and I agree it will help in reading